dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb

```diff
@@ -0,0 +1,52 @@
+require_relative '../../../../../../pegasus/forms/pd_workshop_survey'
+
+class Api::V1::Pd::WorkshopSurveyReportController < Api::V1::Pd::ReportControllerBase
+  load_resource :workshop, class: 'Pd::Workshop'
+
+  FACILITATOR_EFFECTIVENESS_QUESTIONS = [:how_much_learned_s, :how_motivating_s, :how_clearly_presented_s, :how_interesting_s, :how_often_given_feedback_s, :help_quality_s, :how_comfortable_asking_questions_s, :how_often_taught_new_things_s]
+  OVERALL_SUCCESS_QUESTIONS = [:more_prepared_than_before_s, :know_where_to_go_for_help_s, :suitable_for_my_experience_s, :would_recommend_s, :part_of_community_s]
+
+  # GET /api/v1/pd/workshops/:id/aggregate_workshop_score
+  def aggregate_workshop_score
+    survey_report = Hash.new(0)
+
+    survey_report[:this_workshop] = get_score_for_workshops([@workshop])
+    survey_report[:all_my_workshops] = get_score_for_workshops(Pd::Workshop.where(organizer_id: @workshop.organizer_id, course: @workshop.course))
+
+    render json: survey_report
+  end
+
+  private
+
+  def get_score_for_workshops(workshops)
+    report_column = Hash.new(0)
+    response_count = 0
+
+    workshops.flat_map(&:enrollments).each do |enrollment|
+      response_object = PEGASUS_DB[:forms].where(source_id: enrollment.id).first
+
+      next if response_object.nil?
```

Comment:
`Forms.source_id` is unique for a given form kind, but not globally (though in practice it is at the moment). To be safe, add `kind: 'PdWorkshopSurvey' to the where clause, or (preferably) you can instead retrieve the id directly from `enrollment.completed_survey_id`, i.e.
```ruby
response_object = PEGASUS_DB[:forms].where(id: enrollment.completed_survey_id).first
```